tripal_chado/src/Plugin/Field/FieldWidget/ChadoSynonymWidgetDefault.php

```diff
@@ -105,12 +105,22 @@ public function massageFormValues(array $values, array $form, FormStateInterface
       $pub_id = $biodb->insert('pub')->fields(['uniquename' => $uniquename,'type_id' => $type_id])->execute();
     }
 
-    // Remove any empty values that aren't mapped to a record id.
+    //Remove any empty values that aren't mapped to a record id.
     foreach ($values as $val_key => $value) {
-      # if i need to change the format for name to be sgml, i can do it here
-      $values[$val_key]['synonym_sgml'] = $value['name'];
-      $values[$val_key]['syn_type_id'] = $type_id;
-      $values[$val_key]['pub_id'] = $pub_id;
+      if ($value['value'] == '' and $value['record_id'] == 0) {
```

Comment:
``` if ($value['value'] == '' and $value['record_id'] == 0) {```
should be
``` if ($value['name'] == '' and $value['record_id'] == 0) {```
otherwise the whole thing gets deleted and this is why we can't save the synonym